### ISSUE TITLE
dart2js: switch to the new RTI by default

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/sdk#39732]
We need to flip the default. Implementation is practically done, but we need to triage and fix a few test failures before we can enable it by default.

**Comment by sigmundch** [comment]
CL to flip it and add to CHANGELOG is ready,  but we are fixing a few remaining test failures:
  * missing type information on native types that are folded into the parent chain
  * missing subtype information on some bounds

@fishythefish is currently looking.

**Comment by sigmundch** [comment]
@fishythefish fixed all the remaining issues, and the flag was flipped by default as of d735f1f12d1faf45fcee599f1541f87bfbd6b893

woohoo!
